packages/database/src/prisma-api.ts

```diff
@@ -394,6 +394,26 @@ export function getSdk(
         },
       })
 
+      // if this user has already purchased this product, then an additional
+      // purchase (even for only 1 seat) should be processed as a bulk
+      // purchase so that they can distribute the seat to someone else.
+      //
+      // if an existingPurchase is found, but no existingBulkCoupon is found,
+      // then the logic below will account for that and create a new bulk
+      // coupon for the requested quantity.
+      const existingPurchase = await ctx.prisma.purchase.findFirst({
+        where: {
+          productId,
+          userId,
+          status: 'Valid',
+        },
+      })
+
+      // TODO: This doesn't seem to be looking up the bulk coupon based
+      // on the product ID which will become an issue when any particular
+      // app has more than one product.
+      // E.g. should probably account for `restrictedToProductId`
+      //
```

Comment:
I think we have a gap here that hasn't been an issue for single-product apps, but we need to be a little more specific with the query. I'll deal with this in a separate PR.

Reply:
👍